src/tactic/ui/panel/collection_wdg.py

```diff
@@ -0,0 +1,852 @@
+                    spt.notify.show_message("Nothig selected");
+                    return;
+                }
+                var collection_key = bvr.src_el.getAttribute("collection_key");
+
+                var cmd = "tactic.ui.panel.base_table_layout_wdg.CollectionAddCmd";
+                var kwargs = {
+                    collection_key: collection_key,
+                    search_keys: search_keys
+                }
+                var server = TacticServerStub.get();
+                server.execute_cmd(cmd, kwargs);
+                '''
+            } )
+
+
+        add_div = DivWdg()
+        dialog.add(add_div)
+        icon = IconWdg(name="Add new collection", icon="BS_PLUS")
+        add_div.add(icon)
+        add_div.add("Create new Collection")
+        add_div.add_style("margin: 10px")
+        add_div.add_class("hand")
+
+
+        insert_view = "insert"
+
+        add_div.add_behavior( {
+            'type': 'click_up',
+            'insert_view': insert_view,
+            'cbjs_action': '''
+                var top = bvr.src_el.getParent(".spt_table_top");
+                var table = top.getElement(".spt_table");
+                var search_type = top.getAttribute("spt_search_type")
+                kwargs = {
+                  search_type: search_type,
+                  mode: 'insert',
+                  view: bvr.insert_view,
+                  save_event: bvr.event_name,
+                  show_header: false,
+                  default: {
+                    _is_collection: true,
+                  }
+                };
+                spt.panel.load_popup('Add New Collection', 'tactic.ui.panel.EditWdg', kwargs);
+            '''
+        } )
+
+ 
+
+        """
+        button = ActionButtonWdg(title="Add")
+        button.add_style("margin: 10px")
+        dialog.add(button)
+
+
+        button.add_behavior( {
+            'type': 'click',
+            'cbjs_action': '''
+            var search_keys = spt.table.get_selected_search_keys(false);
+
+            var top = bvr.src_el.getParent(".spt_content_top");
+            var values = spt.api.get_input_values(top);
+            console.log(values);
+
+            '''
+        } )
+        """
+
+        return top
+
+
+
+
+class CollectionAddCmd(Command):
+
+    def execute(my):
+
+        collection_key = my.kwargs.get("collection_key")
+        search_keys = my.kwargs.get("search_keys")
+
+        collection = Search.get_by_search_key(collection_key)
+        if not collection:
+            raise Exception("Collection does not exist")
+
+
+        search_type = collection.get_base_search_type()
+        parts = search_type.split("/")
+        collection_type = "%s/%s_in_%s" % (parts[0], parts[1], parts[1])
+        search = Search(collection_type)
+        search.add_filter("parent_code", collection.get_code())
+        items = search.get_sobjects()
+
+
+        search_codes = [x.get_value("search_code") for x in items]
```

Comment:
This depends on the number of items.  If you are doing a lot of look-ups, for example as in:

if code in search_codes:
   ...

This is a lot faster if search_codes is a set because you get a binary look up rather than linear with an array.  So it has a dual purpose of making it unique and having faster look up.  Does this  performance matter in this case?  Probably not.